main.go

```diff
@@ -71,11 +98,19 @@ func main() {
 		LeaderElection:         enableLeaderElection,
 		LeaderElectionID:       "39f5e5c3.nodefeaturediscoveries.nfd.kubernetes.io",
 	})
+
+	// If the manager could not be started, then log the error as an operator setup error
 	if err != nil {
 		setupLog.Error(err, "unable to start manager")
 		os.Exit(1)
 	}
 
+	// Try to create an NFD reconciler using the manager we just created. Note that we do
+	// create a different logger for the NFD reconciler with the name "NodeFeatureDiscovery"
+	// and "controllers" in the name to let users know which step in the operator setup is
+	// occurring. If this step succeeds, then everything related to controllers will be logged
+	// in reference to  "controllers." If this step fails, everything will be logged to the
+	// "setup" logger defined at the top of this file.
```

Comment:
If this step fails we exit with an error(?) (I think this note is self evident from the code and could be dropped)

Reply:
Dropped this note